apps/meteor/app/api/server/v1/users.ts

```diff
@@ -555,6 +559,129 @@ API.v1.addRoute(
 	},
 );
 
+API.v1.addRoute(
+	'users.listByStatus',
+	{
+		authRequired: true,
+		validateParams: isUsersListStatusProps,
+		permissionsRequired: ['view-d-room', 'view-outside-room'],
```

Comment:
```suggestion
		permissionsRequired: ['view-d-room'],
```
I believe only this permission is required for everyone. `view-outside-room` should be checked conditionally based on the `API_Apply_permission_view-outside-room_on_users-list` settings (as in `users.list`), please add this too

Reply:
Is there a reason for it to be checked specificaly in the body of the function instead of in the permissions array? I remember there was a discussion and at the end it was decided to use it like it is right now.

Reply:
Yep, using the `permissionsRequired` array is better overall
But I _believe_ you don't mean to make the `view-outside-room` permission mandatory for this endpoint in this case. Reason: by checking the `users.list` endpoint, I found out that the `view-outside-room` permission is only checked conditionally (which I guess that is the behavior you'd like to mirror). 
If you keep it in the array, then this permission would be required even if the setting is turned off 

This is what we do in `users.list`:
```js
if (
	settings.get('API_Apply_permission_view-outside-room_on_users-list') &&
	!(await hasPermissionAsync(this.userId, 'view-outside-room'))
) {
	return API.v1.unauthorized();
}
```

Reply:
Makes sense, will fix